WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift

```diff
@@ -224,47 +249,84 @@ class BloggingRemindersFlowSettingsViewController: UIViewController {
         ])
     }
 
-    private func populateCalendarDays() {
-        daysOuterStackView.addArrangedSubviews([daysTopInnerStackView, daysBottomInnerStackView])
+    // MARK: - Calendar Days Buttons
+
+    /// Creates the calendar day toggle buttons.  This is a convenience method to take care of the mapping of the day index, from Apple's calendar, to
+    /// our `BloggingRemindersScheduler.Weekday`.  In theory this should never return `nil`, but we're allowing it to avoid possible crashes.
+    ///
+    /// - Parameters:
+    ///     - weekday: the weekday the button is for.
+    ///
+    /// - Returns: the requested toggle button.
+    ///
+    private func createCalendarDayToggleButton(dayIndex: Int) -> CalendarDayToggleButton? {
+        let localizedDayIndex = (dayIndex + calendar.firstWeekday - 1) % calendar.shortWeekdaySymbols.count
+
+        guard let weekday = BloggingRemindersScheduler.Weekday(rawValue: localizedDayIndex) else {
+            return nil
+        }
+
+        let isSelected = weekdays.contains(weekday)
+
+        return CalendarDayToggleButton(
+            weekday: weekday,
+            dayName: self.calendar.shortWeekdaySymbols[localizedDayIndex].uppercased(),
+            isSelected: isSelected) { button in
```

Comment:
We need a weak self here, otherwise we have a retain cycle between the buttons and the VC and neither gets released.

Reply:
Good catch!